File: leetcode/q437/n.py
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def traverse(self, root: TreeNode, targetSum: int):
        if root is None:
            return 0
        l = self.traverse(root.left, targetSum)
        r = self.traverse(root.right, targetSum)
        return l + r + self.countSum(root, targetSum)

    def countSum(self, node: TreeNode, targetSum: int) -> int:
        if node is None:
            return 0
        now = 1 if node.val==targetSum else 0
        return now + self.countSum(node.left, targetSum - node.val) + self.countSum(node.right,
                targetSum-node.val)

    def pathSum(self, root: TreeNode, targetSum: int) -> int:
        return self.traverse(root, targetSum)

File: nowcoder/nc15/n.cpp
#include<iostream>
#include<vector>
#include<cstdio>
#include<cstring>
#include<algorithm>
#include<queue>

using namespace std;

struct TreeNode {
	int val;
 	struct TreeNode *left;
 	struct TreeNode *right;
};

class Solution {
public:
    vector<vector<int> > levelOrder(TreeNode* root) {
        if(root == NULL)
            return {};
        vector<vector<int>> ans;
        queue<TreeNode*> q;
        q.push(root);
        while(q.size()) {
            int num = q.size();
            vector<int> level;
            while(num--) {
                TreeNode* node = q.front();
                q.pop();
                level.push_back(node->val);
                if(node->left)
                    q.push(node->left);
                if(node->right)
                    q.push(node->right);
            }
            if(level.size())
                ans.push_back(level);
        }
        return ans;
    }
};
